src/components/settings/SettingsAppearanceTab/CustomThemeModal.jsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+  ActionRow, Button, Form, Hyperlink, Icon, ModalDialog,
+} from '@edx/paragon';
+import { InfoOutline } from '@edx/paragon/icons';
+import { HELP_CENTER_BRANDING_LINK } from '../data/constants';
+
+export default function CustomThemeModal({
+  isOpen, close, customColors, setTheme,
+}) {
+  const [button, setButton] = useState('');
+  const [buttonValid, setButtonValid] = useState(true);
+  const [banner, setBanner] = useState('');
+  const [bannerValid, setBannerValid] = useState(true);
+  const [accent, setAccent] = useState('');
+  const [accentValid, setAccentValid] = useState(true);
+
+  const headerText = 'Customize the admin and learner edX experience using your own brand colors. Enter color values in hexadecimal code.';
+  const infoText = 'More details about color appearance in the admin and learner experiences and best practices for selecting accessible colors are available in the ';
+  const invalidMessage = 'Must be hexidecimal starting with # (Ex: #1e0b57)';
+  const validHex = new RegExp('^#([A-Fa-f0-9]{6})$');
+
+  function validateColor(input, setter) {
+    if (!input.match(validHex)) {
+      setter(false);
+    } else {
+      setter(true);
+    }
+  }
+
+  function setCustom() {
+    const CUSTOM_THEME = {
+      title: 'Custom Theme',
+      button: (button === '' ? customColors?.button : button),
+      banner: (banner === '' ? customColors?.banner : banner),
+      accent: (accent === '' ? customColors?.accent : accent),
+    };
+    setTheme([CUSTOM_THEME, CUSTOM_THEME]);
+    close();
+  }
+
+  return (
+    <ModalDialog
+      title="My dialog"
+      isOpen={isOpen}
+      onClose={close}
+      size="md"
+      hasCloseButton
+      isFullscreenOnMobile
+    >
+      <ModalDialog.Header>
+        <ModalDialog.Title>
+          Custom Theme
+        </ModalDialog.Title>
+        <p className="mt-3">{headerText}</p>
+      </ModalDialog.Header>
+
+      <ModalDialog.Body>
+        <Form>
```

Comment:
nit: btw, you should check out `Formik`. it's used by quite a few MFEs now for implementing forms to avoid having to write all those `useState` hooks yourself. We recently introduced it in frontend-app-learner-portal-enterprise as well in the `UserEnrollmentForm` component :) 10/10 would recommend again 😹 

https://formik.org/docs/overview

Reply:
Ok I think this is a good idea, but also I kind of want to make a ticket to do this will all forms in admin-portal....